src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -770,8 +770,10 @@ double ThermalCluster::getMarginalCost(uint serieIndex, uint hourInTheYear) cons
      inside -> that is used for all (e.g.10) TS*/
 }
 
-double ThermalCluster::getMarketBidCost(uint serieIndex, uint hourInTheYear) const
+double ThermalCluster::getMarketBidCost(uint hourInTheYear, uint year) const
 {
+    uint serieIndex = (series->timeSeries.width == 1) ? 0 : series->timeseriesNumbers[0][year];
```

Comment:
A small drawback of the new approach is that it makes us compute that indirection at every call.

We could provide a wrapper class to do it only once instead, which would expose a `getMarketBidCost`, something like:
```cpp
// wrapper for the data of one particular year
class ThermalClusterYearData
{
public:
   ThermalClusterYearData(ThermalCluster cluster, uint year);
    double getMarketBidCost(uint hour);
};
//created from the cluster class:
class ThermalCluster
{
public:
    ThermalClusterYearData getYearData(uint year);
}
```

Not a MUST though.